oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/OAuthRequest.java

```diff
@@ -71,15 +97,62 @@ public OAuthRequest(HttpServletRequest request)
 		String token = this.message.getToken();
 		if (token != null) {
 			this.accessor.tokenSecret = OAuthConfiguration.getInstance()
-					.getTokenStrategy().getTokenSecret(this.httpRequest, token);
+					.getJaxTokenStrategy().getTokenSecret(token);
+		}
+	}
+
+	public static class OAuthServletRequestWrapper extends HttpServletRequestWrapper {
```

Comment:
add comment why to do this 

> If the parameter data was sent in the request body, such as occurs with an HTTP POST request, then reading the body directly via getInputStream() or getReader() can interfere with the execution of this method.

https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)